src/main/java/frc/trigon/robot/subsystems/turret/kablamaturret/KablamaTurretConstants.java

```diff
@@ -0,0 +1,68 @@
+package frc.trigon.robot.subsystems.turret.kablamaturret;
+
+import com.ctre.phoenix6.StatusSignal;
+import com.ctre.phoenix6.configs.CANcoderConfiguration;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.hardware.CANcoder;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.AbsoluteSensorRangeValue;
+import com.ctre.phoenix6.signals.InvertedValue;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import com.ctre.phoenix6.signals.SensorDirectionValue;
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+
+public class KablamaTurretConstants {
+    private static final int
+            MOTOR_ID = 1,
+            ENCODER_ID = 2;
+    private static final InvertedValue INVERTED_VALUE = InvertedValue.CounterClockwise_Positive;
+    private static final NeutralModeValue NEUTRAL_MODE_VALUE = NeutralModeValue.Brake;
+    private static final AbsoluteSensorRangeValue ENCODER_RANGE = AbsoluteSensorRangeValue.Signed_PlusMinusHalf;
+    private static final SensorDirectionValue ENCODER_DIRECTION = SensorDirectionValue.CounterClockwise_Positive;
+    private static final double ENCODER_OFFSET = 0;
+    private static final double
+            HUB_X = 8.2296,
+            HUB_Y = 0.5121;
+
+    private static final Translation2d HUB_POSITION = new Translation2d(HUB_X, HUB_Y);
+    private static final Rotation2d HUB_ROTATION = Rotation2d.fromRotations(0);
+    private static final CANcoder ENCODER = new CANcoder(ENCODER_ID);
+    static final TalonFX MOTOR = new TalonFX(MOTOR_ID);
+    static final boolean FOC_ENABLED = true;
+    static final StatusSignal<Double> STATUS_SIGNAL = ENCODER.getPosition();
+    static final Pose2d HUB_POSE = new Pose2d(HUB_POSITION, HUB_ROTATION);
+
+    private static final double
+            P = 1,
+            I = 0,
+            D = 0;
+    static final PIDController PID_CONTROLLER = new PIDController(P, I, D);
```

Comment:
Since the CANcoder and the TalonFX are from the same company, you don't need to use a pidcontroller and can just use the onboard pid. This means you set the pid gains in the motor's config and give the motor requests like `PositionVoltage`

Reply:
wdym `positionVoltage`? also, I can't figure out how to actually use the PID

Reply:
PositionVoltage is a request, just like VoltageOut. You can look at your steer code (and ofc use it like taught in the phoenix 6 presentation) 
You need to apply the pid gains to the config as well as specifying the remote sensor.